_source/_posts/2022-07-21-jhipster-ci-cd.md

```diff
@@ -0,0 +1,865 @@
+    verbs: ['list', 'get']
+  - apiGroups: ['apps']
+    resources: ['controllerrevisions']
+    verbs: ['list']
+  - apiGroups: ['extensions', 'apps']
+    resources: ['daemonsets', 'deployments', 'deployments/scale', 'ingresses', 'replicasets', 'statefulsets']
+    verbs:
+      [
+        'create',
+        'delete',
+        'deletecollection',
+        'get',
+        'list',
+        'patch',
+        'update',
+        'watch',
+      ]
+  # These permissions are necessary for halyard to operate. We use this role also to deploy Spinnaker itself.
+  - apiGroups: ['']
+    resources: ['services/proxy', 'pods/portforward']
+    verbs:
+      [
+        'create',
+        'delete',
+        'deletecollection',
+        'get',
+        'list',
+        'patch',
+        'update',
+        'watch',
+      ]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: spinnaker-role-binding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: spinnaker-role
+subjects:
+  - namespace: spinnaker
+    kind: ServiceAccount
+    name: spinnaker-service-account
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: spinnaker-service-account
+  namespace: spinnaker
+```
+
+Then execute the following commands to create the service account in the current context:
+
+```shell
+CONTEXT=$(kubectl config current-context)
+
+kubectl apply --context $CONTEXT \
+    -f ./spinnaker-service-account.yml
+
+
+TOKEN=$(kubectl get secret --context $CONTEXT \
+   $(kubectl get serviceaccount spinnaker-service-account \
+       --context $CONTEXT \
+       -n spinnaker \
+       -o jsonpath='{.secrets[0].name}') \
+   -n spinnaker \
+   -o jsonpath='{.data.token}' | base64 --decode)
+
+kubectl config set-credentials ${CONTEXT}-token-user --token $TOKEN
+
+kubectl config set-context $CONTEXT --user ${CONTEXT}-token-user
+```
+Enable the kubernetes provider in Halyard:
+
+```shell
+hal config provider kubernetes enable
+```
+
+Add the service account to the Halyard configuration:
+
+```shell
+CONTEXT=$(kubectl config current-context)
+
+hal config provider kubernetes account add spinnaker-gke-account \
+    --context $CONTEXT
+```
+
+### Choose the installation environment
+
+The __Distributed installation__ on Kubernetes deploys each Spinnaker's microservice separately to a remote cloud, and it is the recommended environment for zero-downtime updates of Spinnaker.
+
+Select the distributed deployment with:
+
+```shell
+hal config deploy edit --type distributed --account-name spinnaker-gke-account
+```
+
+### Choose a storage service
+
+Spinnaker requires an external storage provider for persisting application settings and configured pipelines. To avoid losing this data, using a hosted storage solution is recommended. For this example, use Google Cloud Storage. Hosting the data externally allows to delete clusters in between sessions, and keep the pipelines once the clusters have been recreated.
```

Comment:
```suggestion
Spinnaker requires an external storage provider for persisting application settings and configured pipelines. To avoid losing this data, using a hosted storage solution is recommended. For this example, we use Google Cloud Storage. Hosting the data externally allows us to delete clusters in between sessions, and keep the pipelines once the clusters have been recreated.
```